### PR TITLE
go vet fixes, make tests pass on Go 1.10

### PR DESCRIPTION
**Pull request by y0ssar1an**
In Go 1.10, [`go test` runs `go vet`](https://golang.org/doc/go1.10#test). This is a problem because `fzf` currently doesn't pass `go vet`. So `fzf` doesn't pass tests with Go 1.10.

I based this off #1218, which vendors all the dependencies. I couldn't run `go vet ./...` or `go test ./...` on Go 1.10 because of missing packages in `vendor/` (for some reason `make` wasn't fetching them all). I recommend reviewing #1218 first. That will remove a lot of noise from this PR.

### go vet fixes
- add `String()` methods to `Token`, `term`, `Delimeter`, and `*util.Chars` so they can be printed with `%s`
- replace `%s` with `%v` where the default string representation is good enough

### CODE REVIEW

File: src/ansi_test.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func TestExtractColor(t *testing.T) {
 		output, ansiOffsets, newState := extractColor(src, state, nil)
 		state = newState
 		if output != "hello world" {
-			t.Errorf("Invalid output: %s %s", output, []rune(output))
+			t.Errorf("Invalid output: %s %v", output, []rune(output))
```
Comment (y0ssar1an) [inline src/ansi_test.go]: `%s` doesn't work with `[]rune`.
`%v` will print `[]rune("hello")` like this: `[]rune{'h', 'e', 'l', 'l', 'o'}`.

Comment (y0ssar1an) [inline src/ansi_test.go]: Apparently, you can't use `%s` with a rune slice.
`%v` will print `[]rune("hello")` like `[]rune{'h', 'e', 'l', 'l', 'o'}`.

File: src/options_test.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func TestDelimiterRegexString(t *testing.T) {
 		tokens[2].text.ToString() != "---*" ||
 		tokens[3].text.ToString() != "*" ||
 		tokens[4].text.ToString() != "---" {
-		t.Errorf("%s %s %d", delim, tokens, len(tokens))
+		t.Errorf("%s %v %d", delim, tokens, len(tokens))
```
Comment (y0ssar1an) [inline src/options_test.go]:
```go
tokens := []Token{
	Token{text: &Chars{}},
}
fmt.Printf("%v", tokens) // prints [Token{text: Chars{slice: []byte(""), inBytes: false, trimLengthKnown: false, trimLength: 0, Index: 0}, prefixLength: 0}]
```

File: src/options_test.go [hunk 3]
```diff
@@ -71,7 +71,7 @@ func TestSplitNth(t *testing.T) {
 		if len(ranges) != 1 ||
 			ranges[0].begin != rangeEllipsis ||
 			ranges[0].end != rangeEllipsis {
-			t.Errorf("%s", ranges)
+			t.Errorf("%v", ranges)
```
Comment (y0ssar1an) [inline src/options_test.go]:
```go
r := []Range{
	{1, 2},
	{3, 4},
}
fmt.Printf("%v", r) // prints [{1 2} {3 4}]
```

File: src/pattern.go [hunk 4]
```diff
@@ -34,6 +35,11 @@ type term struct {
 	caseSensitive bool
 }
 
+// String returns the string representation of a term.
+func (t term) String() string {
+	return fmt.Sprintf("term{typ: %d, inv: %v, text: []rune(%q), caseSensitive: %v}", t.typ, t.inv, string(t.text), t.caseSensitive)
+}
```
Comment (y0ssar1an) [inline src/pattern.go]: String representations are kind of an art. I tried to follow the Go syntax representation here. Please let me know if you prefer another format. I'm happy to change it.

**Comment by junegunn** [comment]
Cherry-picked the commit that fixes go vet issue. Thanks.

**Pull request closed without merging**